pom.xml

```diff
@@ -7,7 +7,7 @@
   <groupId>org.passay</groupId>
   <artifactId>passay</artifactId>
   <packaging>jar</packaging>
-  <version>1.6.5-SNAPSHOT</version>
+  <version>1.7.0-SNAPSHOT</version>
```

Comment:
I believe this version bump is unnecessary. The API change you made does not break binary API compatibility.

Reply:
The following code will not compile in this branch:

    final List<CharacterRule> ruleList = Arrays.asList(
            new CharacterRule(EnglishCharacterData.LowerCase),
            new CharacterRule(EnglishCharacterData.UpperCase));
    final PasswordGenerator gen = new PasswordGenerator();
    gen.generatePassword(10, ruleList)
    
Given that this would presumably be a common way to invoke the password generator, then it's not API compatible and the version bump is warranted. I will admit that I was surprised that `japicmp-maven-plugin` didn't complain, but I did not investigate.

Reply:
One day I'll remember that generics are invariant. Created a PR to preserve API compatibility:
https://github.com/vt-middleware/passay/pull/155

(japicmp runs in the verify phase, so you might not have exercised it.)

Reply:
Accepted, thanks!